framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -113,39 +123,46 @@ public ValueTransfer(CFAbstractAnalysis<CFValue, CFStore, CFTransfer> analysis)
         return ValueAnnotatedTypeFactory.getCharValues(intAnno);
     }
 
+    /** Returns a list of possible values, or null if no estimate is available. */
     private List<? extends Number> getNumericalValues(
             Node subNode, TransferInput<CFValue, CFStore> p) {
         CFValue value = p.getValueOfSubNode(subNode);
-        AnnotationMirror numberAnno =
+        List<? extends Number> values = null;
+        AnnotationMirror intValAnno =
                 AnnotationUtils.getAnnotationByClass(value.getAnnotations(), IntVal.class);
-        List<? extends Number> values;
-        if (numberAnno == null) {
-            numberAnno =
-                    AnnotationUtils.getAnnotationByClass(value.getAnnotations(), DoubleVal.class);
-            if (numberAnno != null) {
-                values =
-                        AnnotationUtils.getElementValueArray(
-                                numberAnno, "value", Double.class, true);
-            } else {
-                return new ArrayList<Number>();
-            }
-        } else {
-            values = AnnotationUtils.getElementValueArray(numberAnno, "value", Long.class, true);
+        if (intValAnno != null) {
+            values = AnnotationUtils.getElementValueArray(intValAnno, "value", Long.class, true);
+        }
+        AnnotationMirror doubleValAnno =
+                AnnotationUtils.getAnnotationByClass(value.getAnnotations(), DoubleVal.class);
+        if (doubleValAnno != null) {
+            values =
+                    AnnotationUtils.getElementValueArray(
+                            doubleValAnno, "value", Double.class, true);
+        }
+        if (values == null) {
+            return null;
         }
         return NumberUtils.castNumbers(subNode.getType(), values);
     }
 
     private AnnotationMirror createStringValAnnotationMirror(List<String> values) {
-        if (values.isEmpty()) {
+        if (values == null) {
             return ((ValueAnnotatedTypeFactory) atypefactory).UNKNOWNVAL;
         }
+        if (values.isEmpty()) {
+            return ((ValueAnnotatedTypeFactory) atypefactory).BOTTOMVAL;
+        }
```

Comment:
I got rid of all this checks in typetools/checker-framework#994 because ValueAnnotatedTypeFactory#createStringAnnotation is doing these checks already. Do we still need to keep them?